<center>
<h3>Universidade Federal do Rio Grande do Sul (UFRGS)<br>
Programa de Pós-Graduação em Engenharia Civil (PPGEC)<br>
PEC00144 - Métodos Experimentais em Engenharia Civil</h3>

<h1>Trabalho 2</h1>

<h3>Aluno: <a href="https://github.com/dutitello/">Eduarto Pagnussat Titello</a></h3>

</center>




Este trabalho tem por objetivo: no tema de interesse, adotar um problema imaginário de formulação explícita, com 2 ou mais parâmetros dimensionais. Desenhar, formular e definir uma nova escala para o problema, fazendo uma análise de mudança de escala de forma a poder construir tal modelo.

In [1]:
# Importando e configurando módulos
import numpy as np
import pandas as pd 
import matplotlib.pyplot as plt
#import matplotlib
%config InlineBackend.figure_format = 'svg' # Muda backend do jupyter para SVG ;)
import jupyter2latex as j2l # Uma maneira que encontrei para tabelas ficarem ok (github.com/dutitello/Jupyter2Latex)
from math import pi


# Apresentação do problema

O problema adotado consiste na avaliação das duas primeiras frequências naturais de vibração de um poste em concreto armado, considerando seu comportamento como elástico linear. O poste adotado tem comprimento efetivo $L= 12m$, seção constante de diâmetro externo $d_e = 50cm$ e espessura $e=6 cm$, resultando em um diâmetro interno $d_i = 38cm$, conforme a figura:



![Poste](Poste.png)

Conforme Rocha (2020a), as frequências naturais de vibração de uma viga engastada livre (modelo adotado para o poste) podem ser obtidas por \ref{eq:wn}:

\begin{equation}
f_n = \frac{1}{2\pi} {\Big( \frac{\alpha_n}{L} \Big)}^2 \sqrt{\frac{EI}{\mu_L}}
\label{eq:wn}
\end{equation}

onde, para as duas primeiras frequências naturais, $\alpha_1=1.88$ e $\alpha_2=4.69$, respectivamente, e $\mu_L$ é a massa linear da estrutura (massa/comprimento).

A área da seção transversal circular vazada $A$ e seu momento de inércia $I$ são dados por:

\begin{equation}
A = \frac{\pi}{4}\big(d_e^2 - d_i^2 \big)
\label{eq:Ac}
\end{equation}

\begin{equation}
I = \frac{\pi}{64}\big(d_e^4 - d_i^4 \big)
\label{eq:Ic}
\end{equation}



Adotando o subíndice "$_c$" para representar a estrutura real em concreto, sendo empregado concreto da classe C25, o módulo de elasticidade do material é $E_c=28 GPa$, enquanto sua densidade é aproximadamente $\rho_c=2500 kg/m^3$. As propriedades da estrutura original são então:

In [2]:
L_c   = 12.0   # m
de_c  = 0.50   # m
di_c  = 0.38   # m
E_c   = 28.0E9 # N/m^2
rho_c = 2500   # kg/m^3

A_c   = pi/4*(de_c**2 - di_c**2)  # m^2
I_c   = pi/64*(de_c**4 - di_c**4) # m^4

EI_c  = E_c*I_c   # Nm^2
muL_c = A_c*rho_c # kg/m
m_c   = muL_c*L_c # kg

print(f'''
Propriedades da estrutura:
    Área da seção             A_c = {A_c:.3f} m^2
    Momento de Inércia        I_c = {I_c:.3E} m^4 
    Rigidez EI             (EI)_c = {EI_c:.3E} N.m^2
    Massa linear             μL_c = {muL_c:.2f} kg/m
    Massa total               m_c = {m_c:.2f} kg
''')


Propriedades da estrutura:
    Área da seção             A_c = 0.083 m^2
    Momento de Inércia        I_c = 2.044E-03 m^4 
    Rigidez EI             (EI)_c = 5.724E+07 N.m^2
    Massa linear             μL_c = 207.35 kg/m
    Massa total               m_c = 2488.14 kg



# Definição do modelo reduzido

Conforme a expressão \ref{eq:wn}, as frequências naturais da estrutura são dependentes do seu comprimento $L$, da sua rigidez à flexão $EI$ e da sua massa linear $\mu_L$. Em relação às dimensões, o fator de escala adotado é 1:25. Adotando um modelo reduzido em alumínio, com módulo de elasticidade $E_m=70GPa$ e densidade $\rho_m=2700kg/m^3$, os demais fatores de escala adotados são relativos à rigidez $EI$ e a massa linear $\mu_L$. 

Os fatores de escala $EI$ e $\mu_L$ devem considerar perfis de alumínio comercialmente viáveis, dessa forma, é adotado o perfil cilindrico cheio VR-002 da Aluita (Aluita, 2019), de diâmetro externo $d_{e,m}=6.00mm$. O perfil cilíndrico cheio é adotado de forma a reduzir as frequências naturais do modelo reduzido, visto que esse possuí razão $EI/\mu_L$ menor que os perfis tubulares da marca (TR-002 e TR-004). 
Os fatores de escala são então obtidos relacionando as propriedades da seção do modelo à seção do poste:

In [3]:
# Perfil tubular TR-004
#de_m  = 12.70/1000 # m
#di_m  = 10.22/1000 # m
# Perfil tubular TR-002
#de_m  = 9.53/1000  # m
#di_m  = 6.35/1000  # m
# Perfil cilíndrico VR-002
de_m  = 6/1000     # m
di_m  = 0/1000     # m
E_m   = 70.0E9     # N/m^2
rho_m = 2700       # kg/m^3

A_m   = pi/4*(de_m**2 - di_m**2)  # m^2
I_m   = pi/64*(de_m**4 - di_m**4) # m^4
EI_m  = E_m*I_m    # Nm^2
muL_m = A_m*rho_m  # kg/m

print(f'''
Propriedades do modelo reduzido:
    Área da seção             A_m = {A_m:.3E} m^2
    Momento de Inércia        I_m = {I_m:.3E} m^4 
    Rigidez EI             (EI)_m = {EI_m:.3E} N.m^2
    Massa linear             μL_m = {muL_m:.2E} kg/m
''')


Propriedades do modelo reduzido:
    Área da seção             A_m = 2.827E-05 m^2
    Momento de Inércia        I_m = 6.362E-11 m^4 
    Rigidez EI             (EI)_m = 4.453E+00 N.m^2
    Massa linear             μL_m = 7.63E-02 kg/m



In [4]:
escala_L   = 1/25
escala_muL = muL_m/muL_c
escala_EI  = EI_m/EI_c

print(f'''
Fatores de escala impostos:
    Comprimento      L = 1:{1/escala_L:.3E} 
    Massa linear    μL = 1:{1/escala_muL:.3E} 
    Rigidez         EI = 1:{1/escala_EI:.3E}
''')


Fatores de escala impostos:
    Comprimento      L = 1:2.500E+01 
    Massa linear    μL = 1:2.716E+03 
    Rigidez         EI = 1:1.285E+07



# Análise do modelo reduzido 

A interpretação dos resultados do modelo reduzido requer que sua resposta seja escalada de forma à corresponder à estrutura real. Para obtenção de tal fator de escala, a base fundamental do sistema de unidades deve corresponder aos dimensionais das escalas impostas, dessa forma o sistema $L, M, T$ (comprimento, massa, tempo) é inicialmente transformado em um sistema de base $L, \mu_L, EI$ (comprimento, massa linear, rigidez EI), denominado no código como ABC. Fazendo uso da metodologia implementada em Rocha (2020b):

In [5]:
# Importando DimData 
DimData = pd.read_excel('../resources/DimData.xlsx', 
                         index_col  =  0,
                         sheet_name = 'DimData')


# Unidades fundamentais originais 
LMT = ['L', 'M', 'T']

# Novas unidades fundamentais
ABC = ['L', 'μL', 'EI'] 

# Importa matriz dimensional de ABC na base LMT
base = DimData.loc[ABC, LMT]
j2l.df2table(base, 'L, μL, EI na base L, M, T')

# Inverte base de unidades de LMT para ABC
base_i = pd.DataFrame(np.linalg.inv(base), index=LMT, columns=ABC)
j2l.df2table(base_i, 'L, M, T na base L, μL, EI')

L, μL, EI na base L, M, T 
 
 
 
 
 L 
 M 
 T 
 
 
 
 
 L 
 1 
 0 
 0 
 
 
 μL 
 -1 
 1 
 0 
 
 
 EI 
 3 
 1 
 -2

L, M, T na base L, μL, EI 
 
 
 
 
 L 
 μL 
 EI 
 
 
 
 
 L 
 1.0 
 0.0 
 0.0 
 
 
 M 
 1.0 
 1.0 
 0.0 
 
 
 T 
 2.0 
 0.5 
 -0.5

Conforme \ref{eq:wn}, os parâmetros dimensionais relacionados as frequências naturais de vibração $f$ são $L$, $\mu_L$ e $EI$, sendo que $L$, $\mu_L$ e $EI$ são impostos, enquanto $f$ é mantido livre. Formando a matriz dimensional $\bf D$ do problema no sistema ABC, considerando ainda, a titulo de verificação, a massa $m$, temos que: 

In [6]:
par = ['f', 'L', 'μL', 'EI', 'm']
npar = len(par)

DMat_LMT = DimData.loc[par, LMT]
DMat_ABC = np.matmul(DMat_LMT, base_i)
DMat_ABC.rename(columns=dict(zip(LMT, ABC)), inplace=True) # Renomeia colunas para nova base
j2l.df2table(DMat_ABC, 'Matriz D na base L, μL, EI')

Matriz D na base L, μL, EI 
 
 
 
 
 L 
 μL 
 EI 
 
 
 
 
 f 
 -2.0 
 -0.5 
 0.5 
 
 
 L 
 1.0 
 0.0 
 0.0 
 
 
 μL 
 0.0 
 1.0 
 0.0 
 
 
 EI 
 0.0 
 0.0 
 1.0 
 
 
 m 
 1.0 
 1.0 
 0.0

Através da matriz dimensional $\bf D$ no sistema ABC os parâmetros de escala podem ser calculados:

In [7]:
# Ok, não é "bonito" isso de ficar fazendo escala=f(escala)...
escalas = np.array([escala_L, escala_muL, escala_EI])
escalas = np.tile(escalas, (npar, 1))
escalas = np.prod(escalas**DMat_ABC, axis=1)
escalas = pd.DataFrame({'λ': escalas, '1/λ': 1/escalas}, index=par)
j2l.df2table(escalas, 'Fatores de escala')

Fatores de escala 
 
 
 
 
 λ 
 1/λ 
 
 
 
 
 f 
 9.084917e+00 
 1.100726e-01 
 
 
 L 
 4.000000e-02 
 2.500000e+01 
 
 
 μL 
 3.681818e-04 
 2.716049e+03 
 
 
 EI 
 7.779366e-08 
 1.285452e+07 
 
 
 m 
 1.472727e-05 
 6.790123e+04

Observa-se então que as escalas impostas estão de acordo, e, que as frequências naturais de vibração do modelo reduzido são 9.08 vezes maiores que as da estrutura, enquanto sua massa é 1/67901 da massa do poste original. 

# Validação do modelo reduzido

A validade do modelo em escala pode ser verificada pela expressão \ref{eq:wn}. Para isso o comprimento $L_m$ do modelo reduzido deve ser determinado, enquanto as demais propriedades foram determinadas anteriormente. 

In [8]:
L_m = L_c * escalas.loc['L', 'λ']
print(f'Comprimento do modelo reduzido  L_m = {L_m:.3f} m')

Comprimento do modelo reduzido  L_m = 0.480 m


Aplicando os parâmetros do modelo reduzido na função analítica e a correspondente correção na escala de frequências:

In [9]:
def fn(L, EI, muL):
    fns = 1/(2*pi) * 1/L**2 * (EI/muL)**0.5
    fn1 = 1.88**2 * fns
    fn2 = 4.69**2 * fns
    return fn1, fn2

fn_m = fn(L_m, EI_m, muL_m)
fn_m_esc = np.array(fn_m) * escalas.loc['f', '1/λ']

print(f'''
Frequências naturais de vibração do modelo reduzido:
    f_1 = {fn_m[0]:.3f} Hz
    f_2 = {fn_m[1]:.3f} Hz

Frequências naturais de vibração do poste:
    (obtidos através do fator de escala 1:{escalas.loc['f', '1/λ']:.3f} ou {escalas.loc['f', 'λ']:.3f}:1)
    f_1 = {fn_m_esc[0]:.3f} Hz
    f_2 = {fn_m_esc[1]:.3f} Hz
''')





Frequências naturais de vibração do modelo reduzido:
    f_1 = 18.647 Hz
    f_2 = 116.049 Hz

Frequências naturais de vibração do poste:
    (obtidos através do fator de escala 1:0.110 ou 9.085:1)
    f_1 = 2.053 Hz
    f_2 = 12.774 Hz



Aplicando agora os parâmetros da estrutura real na função analítica:

In [10]:
fn_c = fn(L_c, EI_c, muL_c)

print(f'''
Frequências naturais de vibração do poste:
    f_1 = {fn_c[0]:.3f} Hz
    f_2 = {fn_c[1]:.3f} Hz
''')



Frequências naturais de vibração do poste:
    f_1 = 2.053 Hz
    f_2 = 12.774 Hz



Logo o modelo em escala representa corretamente as frequências do poste.

Visto que escala de massa linear $\mu_L$ foi imposta, a massa do modelo reduzido é a massa de um trecho de tamanho $L_m$ do perfil de alumínio adotado, aplicando sobre essa a escala de massa calculada $\lambda_m$, a massa do poste real deve então ser encontrada, conforme:

In [11]:
m_m     = muL_m * L_m
m_m_esc = m_m * escalas.loc['m', '1/λ']

print(f'''
Comparação das massas:
    Massa do poste                          m_c = {m_c:.3f} kg
    Massa do modelo reduzido                m_m = {m_m:.3f} kg
    Massa escalada do modelo reduzido   m_m_esc = {m_m_esc:.3f} kg
''')


Comparação das massas:
    Massa do poste                          m_c = 2488.141 kg
    Massa do modelo reduzido                m_m = 0.037 kg
    Massa escalada do modelo reduzido   m_m_esc = 2488.141 kg



Logo o modelo em escala está corretamente definido.

# Referências Bibliográficas

Aluita - Alumínio Porto Alegre Ltda. **Catálogo de Perfis 2019/2020**, 2020. Disponível em: <https://aluita.com.br/wp-content/uploads/2020/01/Cat%C3%A1logo-Perfil-2019.pdf>. 

Rocha, M. M. **PEC00025 - Introduction to Vibration Theory**, 2020a. Disponível em: <https://github.com/mmaiarocha/PEC00025>.

Rocha, M. M. **PEC00144 - Experimental Methods in Civil Engineering**, 2020b. Disponível em: <https://github.com/mmaiarocha/PEC00144>. 
